File: src/test/scala/simulations/ExternalOperatorSearchSimulation.scala
package simulations

import io.gatling.core.Predef._
import io.gatling.core.structure.PopulationBuilder
import io.gatling.http.Predef._
import io.gatling.http.protocol.HttpProtocolBuilder
import scenarios.ExternalOperatorSearchUser
import utils.{Headers, SetUp}

import scala.concurrent.duration._
import scala.language.postfixOps


class ExternalOperatorSearchSimulation extends Simulation {

  val httpConfiguration: HttpProtocolBuilder = http.baseUrl(SetUp.baseURL).headers(Headers.requestHeaders)
    .disableCaching
    .disableWarmUp
    .silentResources
    .perUserNameResolution
    .maxConnectionsPerHostLikeChrome

  val search: PopulationBuilder =
    ExternalOperatorSearchUser.companySearch.inject(atOnceUsers(SetUp.users),
          constantUsersPerSec(SetUp.rampUp) during (SetUp.rampDurationInMin minutes))
  setUp(search)
    .protocols(httpConfiguration)
    .assertions(global.failedRequests.count.is(0))
}

File: src/test/scala/simulations/ExternalOperatorSearchSimulation.java
package simulations;

import static io.gatling.javaapi.core.CoreDsl.*;
import static io.gatling.javaapi.core.OpenInjectionStep.atOnceUsers;
import static io.gatling.javaapi.http.HttpDsl.*;
import static utils.SetUp.*;

import io.gatling.javaapi.core.*;
import io.gatling.javaapi.http.*;
import scenarios.ExternalOperatorSearchUser;

import java.time.Duration;


public class ExternalOperatorSearchSimulation extends Simulation {

    private static final HttpProtocolBuilder httpConfiguration = http
            .baseUrl(baseURL)
            .disableAutoReferer()
            .disableCaching()
            .disableWarmUp()
            .silentResources()
            .perUserNameResolution();


    private final PopulationBuilder search = ExternalOperatorSearchUser.companySearch()
            .injectOpen(
                    atOnceUsers(users),
                    constantUsersPerSec(rampUp).during(Duration.ofMinutes(rampDurationInMin))
            );

    {
        setUp(search)
                .protocols(httpConfiguration)
                .assertions(global().failedRequests().count().is(0L));
    }
}